data-donation-sdk/src/commonMain/kotlin/care/data4life/datadonation/donation/publickeyservice/model/PublicKeys.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+package care.data4life.datadonation.donation.publickeyservice.model
+
+internal data class PublicKeys(
+    val donationService: String,
+    val alp: String
+)
```

Comment:
I;d leave this out, i understand that you want to scope down on the interface what we actually need from the service, but it creates 2 different ways to express the same thing (a bundle of keys) and adds the limitation to being them always 2, when we can easily return a list and filter for service later on.

Reply:
Actually the concepts are different - We have a KeyPair (Private/Public) for the Donor and 2 PublicKeys for the Services.

Reply:
Also the name (ServiceCredentialKey) you mention below is more fitting here then.

Reply:
i never mentioned KeyPairs, i know this is 2 public keys, so your answer doesnt affect my comment. ITs a keychain of some sort with 2 public keys, which is only used as a return value from fetching keys and therefore still not needed if we just return a list of keys (which is the whole concept of this credential service)

Reply:
A list is not safe it regards to the order, this mapping is.

Reply:
no need for order, we can use rawkeys with identifiers

Reply:
In other words, the consuming service (aka crypto), needs to known about the keys and should make filtering? Honestly I think this increases the complexity and decreases the readability on the consumer. Also this would violate Clean Architecture. So what is the benefit of doing that?

Reply:
On a 2nd thought, I think I see your concern (please correct me if I am wrong).
The concept of a KeyChain/KeyRing is by looking over the open PRs and the codebase only a iOS thing. For Android we do not need it and also it is nothing Common can build upon. I think what you actually concerned about is, how to make the mapping of Service to Key transparent to iOS's KeyChain, right? Iff so this is not a concern of this PR, it is one for the PR which raises the Bridging code for the crypto of iOS. The issue can be easily solved by then specialise the generic encrypt method in something like `encryptALP` and `encryptDonationService`, which then are for Android only aliases for a generic encrypt method and the Bridging code for iOS can do its thing.

Reply:
As discussed, my point is more about having to read and understand differences on why different models are used from an outsider perspective (PublicKeys with 2 values, RawKeys with the same values) so using a list of filtered keys instead of a new structure would have been easier to handle (i had to go around the code to find what a publicKeys is by knowing Raw Key already for instance) and maybe a good compromise on being very strict of what gets out of the donation service. 

Since it doesn't look like this would be the end goal of this PR anyways for you, i can only propose another name for it. 
I mentioned KeyRing because i am not a fan of classes with a plural name that dont represent lists, so maybe a `ServicePublicKeyring` class with donationKeyBase64Encoded: String and alpKeyBase64Encoded: String can be more obvious as to whats there. 

I feel the need to add the base64encoded detail because on iOS it helps a lot to remember what to use and how with conversion to be explicit, so maybe might make sense here too

Reply:
done